Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm

```diff
@@ -1991,3 +1994,354 @@ HTTPServer httpServer({
     TestWebKitAPI::Util::run(&done);
     EXPECT_FALSE(navigationFailed);
 }
+
+#if HAVE(ALLOW_PRIVATE_ACCESS_TOKENS_FOR_THIRD_PARTY)
+TEST(WKNavigation, FrameNavigationWithPrivateTokenResponse)
+{
+    using namespace TestWebKitAPI;
+
+    static constexpr auto mainBytes =
+"<script>"
+"navigator.serviceWorker.register('/service-worker.js').then(function(reg) {"
+"    location = '/path4';"
+"});"
+"</script>"_s;
+
+    static constexpr auto serviceWorkerBytes =
+"self.addEventListener(\"fetch\", (event) => {"
```

Comment:
Can we do a fetch in "install" event handler and verify request.isPrivateTokenUsageByThirdPartyAllowed is false.

I would also tend to move the service worker specific tests in its own test.
This might make maintenance of the test easier.